parser/src/tests/errors.rs

```diff
@@ -0,0 +1,72 @@
+use crate::error::ParseError::InvalidInt;
+use crate::{build_parse_test, error::Error, tests::ParseTest};
+
+// SCAN ERRORS
+// ================================================================================================
+
+#[test]
+fn error_invalid_token() {
+    let source = "
+    transition_constraints:
+        enf clk'' = clk + 1";
+    // Only one "'" is allowed at the end of identifier to indicate next row of the column.
+    // Should fail if more than one "'" is present in an identifier
+    let error = Error::ScanError(45..46);
+    build_parse_test!(source).expect_error(error);
+}
```

Comment:
We should be able to simplify this so that the source only contains the expression under test. Since scan errors should appear before parse errors, we should be able to get the `ScanError` without including the section keyword.

I would make this change for all other tests as well - make the source string as small as possible while still including the error

Reply:
Sorry, I should have made it clearer. Since logos has recoverable errors so it doesn't fail while scanning. Since we don't have recoverable errors in lalrpop yet, the parser returns the first error it encounters so if we remove "transition_constraints" in the above, it will return an error there since there's no match for source section keyword.
I've changed the `expect_error` function check for the first failing error while scanning and match against that and hence changed the test to what you suggested to make the tests more atomic.
This should be improved when we have recoverable errors at parser level as well in the future.

Reply:
Ok we can revisit it then, thanks for explaining!